### DISCUSSION TITLE
Typescript: Instanceof 'Element' failing with 'Element is not defined' in node

### DISCUSSION THREAD

**Discussion by ajmas** [xmldom/xmldom#658] [Q&A]
I am trying to test to see if a node is an instance of `Element` in nodejs, but I am getting told:

> Element is not defined

when I do:

```ts
if (childNodes[i] instanceof Element) {
  ...
}
```

Right now I am working around this via:

```ts
if ((childNodes[i] as any).getAttribute) {
  ...
}
```

Where should I be pulling the definition from, for Element?

**Comment by karfau** [comment]
which version of the library are you using?
If you are looking for not just Element, but also other node types, have a look at https://github.com/xmldom/is-dom-node#usage
But there is a pending release in the `next` tag (the upcoming one, which is currently in beta), which directly exports all the DOM node "types", which was done as part of #637 
Once this is released, you can directly import it from the `@xmldom/xmldom` import without any additional path.

If you have any further quesitions, let me know.

**Reply by ajmas** [reply]
I am using 0.8.10 of @xmldom/xmldom

I'll take a look at js-dom-node, but my only concern is not having xpath support. I am using the 'xpath' library for now.

BTW out of curiosity I did try @xmldom/xmldom@next, but unfortunately that doesn't provide the Element definition either. Debating whether I need to add the 'dom' to my typescript config will help.

**Reply by karfau** [reply]
Just to be sure: the error you get about Element not being defined is coming from Typescript or is it a runtime error?
In case of a runtime error adding something to your typescript config will not be able to resolve it.

The library is called `@xmldom/is-dom-node`. (the first letter after the / is an i, not a j)

The version containg the exposed Element has not been published to npm yet. I will try to take care of it this week.

I'm not sure how exactly this is related to xpath, since I never used it myself. But I know there are some people that made it work for them. Search for xpath in the closed issues.

**Reply by karfau** [reply]
update: #717 will resolve the type errors around instanceof and will be published as 0.9.1 once it's merged.
